internal/pkg/chainsync/fetcher/graphsync_fetcher.go

```diff
@@ -436,8 +440,13 @@ func (gsf *GraphSyncFetcher) loadAndVerifyFullBlock(ctx context.Context, key blo
 				return err
 			}
 
+			messageCount += len(messages)
 			return nil
 		})
+
+	if messageCount > block.BlockMessageLimit {
+		return block.UndefTipSet, nil, xerrors.New("Number of messages in block exceeds block message limit")
+	}
```

Comment:
I don't think this is the right place to do this. If we had a non-graphsync fetcher, or obtained blocks in any way, we still need to check this. We should be able to validate blocks produced locally.

This should be part of either the syntax or semantic validator (depending whether we give the former access to a store so it can look up the AMT root and `Count` it.) Neither have store access yet, but the semantic one will definitely need it. We currently do that kind of thing in `Expected.ValidateMining`.

Reply:
I'll push a commit moving this to expected to group it with everybody else.